ssl/t1_trce.c

```diff
@@ -1493,6 +1497,8 @@ void SSL_trace(int write_p, int version, int content_type,
     switch (content_type) {
     case SSL3_RT_HEADER:
         {
+            if (msglen < 2)
+                return 0;
             int hvers = msg[1] << 8 | msg[2];
```

Comment:
We don't tolerate declarations in the middle of code. I mean in this case hvers would have to be declared (but not assigned) prior if (msglen). Secondly it looks like you have to compare to 3, not 2. Indeed it's msg[2]. However, if one actually aims to ensure that there is not overstepping, then shouldn't one have to perform more checks even further down?

Reply:
Indeed, pushing a new version of the commit